src/main/kotlin/com/hitc/hitweconomyplugin/main/core/PlayerData.kt

```diff
@@ -1,9 +1,54 @@
 package com.hitc.hitweconomyplugin.main.core
 
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import java.io.File
+
 
 @Serializable
-data class PlayerData(
-    var credits : Int,
-    var creditsEarned : Int
-)
+class PlayerData(
+    private var credits: Int,
```

Comment:
You can init `credits` and `creditsEarned` to 0 so they become optional arguments and so instead of doing `PlayerData(0,0, f)` you can just do `PlayerData(file = f)`